File: Driver.java
import java.io.IOException;
import java.util.ArrayList;
import java.util.Date;
import java.util.HashMap;
import java.util.List;

public class Driver {
    public static <T> void main(String[] args) throws IOException {
        //Number of vertices
        int number_of_v = 32;
        //Isoperimetrical dynamicity
        double isoperimetricDynamicity = (Math.log(number_of_v) / Math.log(2)) / 2;
        //probability
        double probability = (2 * isoperimetricDynamicity) / number_of_v;
        //number of graphs
        int num_graphs = 1000000;

        Write_Graph wgf = new Write_Graph();

        ER_Graph er_gen = new ER_Graph();

        //Write graphs
        for (int i = 0; i < num_graphs; i++) {
        List<ArrayList<Integer>> graph_er = er_gen.generate(probability, number_of_v);
        wgf.write(new String[]{"Test_Folder", "ER"}, "Graphs_ER"+ "_N" + number_of_v + "_V" + num_graphs, graph_er, true);
        }
        System.out.println(wgf.getLocation());

        //Read graphs from file
        Read_Graph rg = new Read_Graph();
        rg.read_graphs(new String[]{"Test_Folder", "ER"}, "Graphs_ER"+ "_N"  + number_of_v + "_V" + num_graphs);
        HashMap<String, List<ArrayList<Integer>>> graphs = rg.getGraphs();
//        System.out.println(graphs);

        //Number of vertices
        int number_of_l = 1;

        //Make new empty Node array
        Node<T>[] Nodes = new Node[number_of_v];
        //Populate array of Nodes via for loop
        boolean isSupervisor;
        for (int i = 0; i < number_of_v; i++) {
            isSupervisor = i < number_of_l;
            Nodes[i] = new Node<>(isSupervisor);
        }


        ISCALA sim = new ISCALA<>();


        //time
        Date date = new Date();

        //run ISCALA
        long start = System.currentTimeMillis();

        long[] output = sim.ISCALA_algo(number_of_l, isoperimetricDynamicity, Nodes, graphs, num_graphs);

        long time = System.currentTimeMillis() - start;

        System.out.println("Time: " + time);
        System.out.println("N: " + output[0]);
        System.out.println("Rounds " + output[1]);
    }
}

File: NewCPP/Driver.cpp
// Driver.cpp
#include <string>
#include <iostream>
#include <vector>
#include <unordered_map>
#include <chrono>
#include <random>
#include "ISCALA.h"
#include "Node.h"
#include "ER_Graph.h"
#include "Read_Graph.h"
#include "Write_Graph.h"
#include "WS_Graph.h"
#include "Star_Graph.h"
#include "RG_Graph.h"
#include "Path_Graph.h"

int main() {
    // Number of vertices
    int number_of_v = 32;
    // Isoperimetrical dynamicity
    double isoperimetricDynamicity = (std::log(number_of_v) / std::log(2)) / 2;
    // Probability
    double probability = (2 * isoperimetricDynamicity) / number_of_v;
    // Number of graphs
    //int num_graphs = 100;
    int num_graphs = 1000000;

    Write_Graph wgf;

    ER_Graph er_gen;
    WS_Graph ws_gen;
    Star_Graph star_gen;
    RG_Graph rg_gen;
    Path_Graph path_gen;

    // Write graphs
    for (int i = 0; i < num_graphs; i++) {
        std::vector<std::vector<int>> graph_er = er_gen.generate(probability, number_of_v);
        wgf.write({"Test_Folder", "ER"}, "Graphs_ER_N" + std::to_string(number_of_v) + "_V" + std::to_string(num_graphs), graph_er, true);
    }
    std::cout << wgf.getLocation() << std::endl;

    // Read graphs from file
    Read_Graph rg;
    rg.read_graphs({"Test_Folder", "ER"}, "Graphs_ER_N" + std::to_string(number_of_v) + "_V" + std::to_string(num_graphs));
    std::unordered_map<std::string, std::vector<std::vector<int>>> graphs = rg.getGraphs();

    // Number of vertices
    int number_of_l = 1;

    // Make new empty Node array
    std::vector<Node<int>> Nodes(number_of_v);
    // Populate array of Nodes via for loop
    for (int i = 0; i < number_of_v; i++) {
        bool isSupervisor = i < number_of_l;
        Nodes[i].setSupervisor(isSupervisor);
    }

    ISCALA<int> sim;

    // Time
    auto start = std::chrono::high_resolution_clock::now();

    // Run ISCALA
    std::vector<long> output = sim.ISCALA_algo(number_of_l, isoperimetricDynamicity, Nodes, graphs, num_graphs);

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end - start;

    std::cout << "Time: " << elapsed.count() << " seconds" << std::endl;
    std::cout << "N: " << output[0] << std::endl;
    std::cout << "Rounds: " << output[1] << std::endl;

    return 0;
}